### PR TITLE
Switch to use the new way for authentication in Node.js

### PR DESCRIPTION
**Pull request by hzyi-google**

### CODE REVIEW

**Comment by hzyi-google** [comment]
@JustinBeckwith I've never coded in JavaScript before so please let me know if there is any naive mistake.

File: toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip [hunk 1]
```diff
@@ -22,28 +22,11 @@
...
-    {@authFunc_OAuth3L(class)}
-  @case "API_KEY"
-    var {@class.auth.authVarName} = '';  // TODO: Update placeholder with desired API key.
-
-    {@mainFuncBody(class)}
-  @end
-@end
+  const auth = await google.auth.getClient({
+    scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+  }
```
Comment (callmehiphop) [inline toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip]: It looks like this is missing the closing parentheses and a semicolon.

File: toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip [hunk 2]
```diff
@@ -22,28 +22,11 @@
   const {{@class.googleImportVarName}} = require('googleapis');
   var {@class.serviceVarName} = {@class.googleImportVarName}.{@class.serviceTypeName}('{@class.apiVersion}');
```
Comment (callmehiphop) [inline toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip]: I'll let @JustinBeckwith chime in, but it looks like you could simplify the examples by passing in `auth` like this

```js
var adExchangeBuyer = google.adexchangebuyer({
  version: 'v1.4',
  auth
});
```

File: toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip [hunk 3]
```diff
@@ -22,28 +22,11 @@
...
-    {@mainFunc(class)}
-
-    {@authFunc_OAuth3L(class)}
-  @case "API_KEY"
-    var {@class.auth.authVarName} = '';  // TODO: Update placeholder with desired API key.
-
-    {@mainFuncBody(class)}
-  @end
-@end
+  const auth = await google.auth.getClient({
```
Comment (callmehiphop) [inline toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip]: Kind of a nitpick, but the examples are mixing callbacks and promises (via `await`). Seems like one or the other should be used.

Also, there currently is no top level support for `await`, so you would need to wrap this in an `async` function.

File: toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_cloudbilling.v1.json.baseline [hunk 4]
```diff
@@ -66,48 +54,36 @@ function authorize(callback) {
 const {google} = require('googleapis');
 var cloudbilling = google.cloudbilling('v1');
 
-authorize(function(authClient) {
-  var request = {
-    auth: authClient,
-  };
-
-  var handlePage = function(err, response) {
```
Comment (hzyi-google) [inline toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_cloudbilling.v1.json.baseline]: @callmehiphop in the original email from @JustinBeckwith it seems `await` is preferred:

> const auth = await google.auth.getClient({
>   scopes: ['https://www.googleapis.com/auth/cloud-platform'']
> });
> google.options({auth});
> const compute = google.compute('v1');
> const response = await compute.instances.list();
> const instances = response.data.items;

Then how should I change the way paged streaming methods are handled?

**Comment by vchudnov-g** [comment]
What are we blocked on here? (Your last commit says "test not pass")

File: toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip [hunk 5]
```diff
@@ -20,30 +20,19 @@
...
+    scopes: ['https://www.googleapis.com/auth/cloud-platform'],
+  });
 
-@private mainFunc(class)
-  {@class.auth.authFuncName}(function({@class.auth.authVarName}) {
-    {@mainFuncBody(class)}
+  var {@class.serviceVarName} = {@class.googleImportVarName}.{@class.serviceTypeName}(
```
Comment (JustinBeckwith) [inline toolkit/src/main/resources/com/google/api/codegen/nodejs/sample.snip]: Can we use `const` here?

File: toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_adexchangebuyer.v1.4.json.baseline [hunk 6]
```diff
@@ -10,39 +10,26 @@
...
-    if (err) {
-      console.error(err);
-      return;
-    }
-
-    // TODO: Change code below to process the `response` object:
-    console.log(JSON.stringify(response, null, 2));
-  });
-});
+const auth = await google.auth.getClient({
```
Comment (JustinBeckwith) [inline toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_adexchangebuyer.v1.4.json.baseline]: looks like you use `auth` for the variable name here, but expect `authClient` below

File: toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_adexchangebuyer.v1.4.json.baseline [hunk 7]
```diff
@@ -10,39 +10,26 @@
...
-  //
-  // Authorize using the following scope:
-  //   'https://www.googleapis.com/auth/adexchange.buyer'
-  var authClient = null;
+  auth: authClient,
+};
 
-  if (authClient == null) {
-    console.log('authentication failed');
+adExchangeBuyer.accounts.get(request, function(err, response) {
```
Comment (JustinBeckwith) [inline toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_adexchangebuyer.v1.4.json.baseline]: kinda to @callmehiphop point - this is a mix of `async` syntax and `await`.  Could we change this to be:

```js
const response = await adExchangeBuyer.accounts.get(request);
```

You can actually drop all the other stuffs this way 🙃

File: toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_adexchangebuyer.v1.4.json.baseline [hunk 8]
```diff
@@ -10,39 +10,26 @@
...
+  if (err) {
+    console.error(err);
     return;
   }
-  callback(authClient);
-}
+
+  // TODO: Change code below to process the `response` object:
+  console.log(JSON.stringify(response, null, 2));
```
Comment (JustinBeckwith) [inline toolkit/src/test/java/com/google/api/codegen/testdata/discoveries/nodejs/nodejs_adexchangebuyer.v1.4.json.baseline]: I think `console.log(response.data)` here would be just fine

**Comment by bcoe** [comment]
@hzyi-google I missed this PR and landed a small patch that at least makes the docs completely incorrect:

https://github.com/googleapis/discovery-artifact-manager/pull/117

Excited to see this more thorough refactor land, anything I can do to help unblock.

**Pull request closed without merging**